storm/src/main/storm/mesos/util/PrettyProtobuf.java

```diff
@@ -6,62 +6,108 @@
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.OfferID;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.SlaveID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.Protos.TaskState;
+import org.apache.mesos.Protos.TaskStatus;
 import org.apache.mesos.Protos.Value.Range;
-import org.apache.mesos.Protos.Value.Ranges;
-import org.apache.mesos.Protos.Value.Set;
 import org.json.simple.JSONValue;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 
 /**
  * This utility class provides methods to improve logging of Mesos protobuf objects.
  * These methods don't perform any logging directly, instead they return Strings
  * which can be logged by callers.
- *
+ * <p/>
  * The methods offer more concise and readable String representations of protobuf
  * objects than you get by calling a protobuf object's native toString method.
- *
+ * <p/>
  * Another advantage over standard protobuf toString() output is that this output is
  * in proper JSON format, resulting in logs that can be more easily parsed.
- * This desire to print in JSON is why we populate Maps below for each multi-field
- * object. Also, we use LinkedHashMap and TreeMap to ensure the order of the fields
- * in multi-field objects stays consistent. That allows log lines to be visually
- * compared whilst examining the state of the system.
- *
+ * <p/>
  * TODO(erikdw):
- *  1. Check whether a value is set/null when deciding whether to include it.
- *  2. Currently we only include the object values that we care about for a
- *     storm-only mesos-cluster.  We should instead allow configuration for choosing
- *     which fields to include (e.g., a bitmap toggling certain fields on/off).
- *  3. Allow cleanly logging to separate Logger, to allow configuring the logs going to
- *     a separate log file.  The complication is that these methods lack context,
- *     they are just pretty-printing the protobuf objects.
+ * 1. Check whether a value is set/null when deciding whether to include it.
+ * 2. Currently we only include the object values that we care about for Groupon's
+ * storm-only mesos-cluster.  We should instead allow configuration for choosing
+ * which fields to include (e.g., a bitmap toggling certain fields on/off).
+ * 3. Allow cleanly logging to separate Logger, to allow configuring the logs going to
+ * a separate log file.  The complication is that these methods lack context,
+ * they are just pretty-printing the protobuf objects.
  */
 public class PrettyProtobuf {
 
   /**
+   * Ideally we'd have a generic method for these getTrimmedString methods, but
+   * these protobuf types don't implement a common interface (which would need
+   * to include getValue()).
+   */
+
+  /**
+   * Pretty-print the mesos protobuf OfferID.
+   */
+  public static String getTrimmedString(OfferID id) {
```

Comment:
I'm not sure if we need `getTrimmedString`.  I'm pretty sure the reason it's not here in storm-mesos is that @tnachen pointed out to me in the [initial review for the logging improvements](https://github.com/mesos/storm/pull/32) that it was unnecessary.  But I left it in place in our internal implementation.  Should be tested.